tfe/resource_tfe_workspace.go

```diff
@@ -231,6 +232,7 @@ func resourceTFEWorkspaceRead(d *schema.ResourceData, meta interface{}) error {
 	if workspace.VCSRepo != nil {
 		vcsConfig := map[string]interface{}{
 			"identifier":         workspace.VCSRepo.Identifier,
+			"branch":             workspace.VCSRepo.Branch,
```

Comment:
sort of the same question here as the one above. i think the goal is to only set the branch if we actually get one back  from the API request and it isn't `""`. there's clearly something wrong with the way we were attempting to do this before but i'm not sure what

Reply:
In this case we should remove the if block below. Read()'s result should not be affected by the HCL config at all, instead it should be reporting what the resource actually looks like. 

I'll update this PR shortly!